### ISSUE TITLE
Expansion to \PackageError in bigintcalc macros

### ISSUE THREAD

**Issue by mikkelee** [ho-tex/bigintcalc#1]
I'm writing a package where I do some complicated lookups to determine which numbers to use in some simple arithmetic. However, in some cases, it may not be possible to look up which number to use, so I put a `\PackageError` there to alert the user.

However, this breaks in an unreadable way. It took me a long time to debug, because all context around the error happened was lost. Below is a much boiled down MWE that shows the problem (the bad error happens in the final use of `\ bigintcalcNum`).

Is there a way for the `\bigintcalc...` macros to handle this? Perhaps a way to check if a macro expands to valid `bigintcalc` input (if such an idea makes sense, I'm still a novice at the more technical aspects of latex)?

```latex
\documentclass{article}

\usepackage{bigintcalc}

\newcommand\myFancyMacro[1]{%
  \ifcase\numexpr#1\relax%
    % case 0:
  	10%
  \or%
    % case 1:
    \undefinedMacro%
  \else%
    \PackageError{test}{Not implemented: #1}{}%
  \fi%
}

\begin{document}

A is OK:
\myFancyMacro{0} = 
\bigintcalcNum{\myFancyMacro{0}} % => 10

B is OK:
\myFancyMacro{1} =  % => Undefined control sequence: \undefinedMacro (as expected)
\bigintcalcNum{\myFancyMacro{1}} % => Undefined control sequence: \undefinedMacro (as expected)

C is broken:
\myFancyMacro{100} = % raises PackageError (as expected)
\bigintcalcNum{\myFancyMacro{100}} % => Undefined control sequence: \GenericError ...

\end{document}
```

**Comment by davidcarlisle** [comment]
bigint can't really do anything here, The error message makes yoru command not safe in an expansion context (`\ref` fails for similar reasons, often you want it to expand for a number but the checking for undefined references makes it non-expandable)

raising  errors in an expansion context is hard, the simplest is simply to use your `\undefinedmacro` approach and not use a real error message `expl3` does have a mechanism for expandable error messages, but is very restricted and not clearly better but try for example


```
\documentclass{article}

\usepackage{bigintcalc}

\ExplSyntaxOn
\newcommand\myFancyMacro[1]{%
  \ifcase\numexpr#1\relax%
    % case 0:
  	10%
  \or%
    % case 1:
    \undefinedMacro%
  \else%
\msg_expandable_error:nn {FancyMacro } { Not ~ Implemented ~ #1 } 
  \fi%
}
\ExplSyntaxOff

\begin{document}

A is OK:
\myFancyMacro{0} = 
\bigintcalcNum{\myFancyMacro{0}} % => 10

B is OK:
\myFancyMacro{1} =  % => Undefined control sequence: \undefinedMacro (as expected)
\bigintcalcNum{\myFancyMacro{1}} % => Undefined control sequence: \undefinedMacro (as expected)

C is broken:
\myFancyMacro{100} = % raises PackageError (as expected)
\bigintcalcNum{\myFancyMacro{100}} % => Undefined control sequence: \GenericError ...

\end{document}
```


(you will need  to add `\usepackage{expl3}` on older latex formats)

This still basically uses an undefined command to trigger an error but allows some restricted messag etect to appear


```
! Undefined control sequence.
<argument> \::error 
                    ! FancyMacro: \msg text > FancyMacro/Not Implemented 100...
```

**Comment by mikkelee** [comment]
Many thanks for the detailed answer, and for the package! 👍
